pkg/lib/bundlev2/synthesis.go

```diff
@@ -0,0 +1,375 @@
+				}
+
+				if bufObj.GetKind() == RoleKind {
+					return b.appendTostructuredObjectMap(RoleKind, bufObj, rv1.Role{}, info.Name())
+				}
+
+				if bufObj.GetKind() == RoleBindingKind {
+					return b.appendTostructuredObjectMap(RoleBindingKind, bufObj, rv1.RoleBinding{}, info.Name())
+				}
+
+				if bufObj.GetKind() == RegistryV2CSVKind {
+					return b.appendTostructuredObjectMap(RegistryV2CSVKind, bufObj, RegistryV2CSV{}, info.Name())
+				}
+
+				if bufObj.GetKind() == DeploymentKind {
+					return b.appendTostructuredObjectMap(DeploymentKind, bufObj, appsv1.Deployment{}, info.Name())
+				}
+
+				// Skipping kinds of resources (PrometheusRule and ServiceMonitor) not needed for the CSV.
+				return nil
+			}
+			b.logger.Warnf("file %s with kind %s is not supported", info.Name(), bufObj.GetKind())
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// appendTostructuredObjectMap converts the unstructured object and try to convert it into a specified interface
+// before appending to the structuredObjectMap. It only returns error if the unstructured is not convertible.
+func (b *bundlev2) appendTostructuredObjectMap(kind string, bufObj unstructured.Unstructured, obj interface{},
+	objSource string) error {
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(bufObj.UnstructuredContent(), &obj); err != nil {
+		return fmt.Errorf("error decoding %s as %s kind, %v", objSource, bufObj.GetKind(), err)
+	}
+
+	if obj, ok := b.structuredObjectMap[kind]; !ok {
+		b.structuredObjectMap[kind] = &[]interface{}{obj}
+	} else {
+		*obj = append(*obj, bufObj)
+	}
+
+	return nil
+}
+
+// getRegistryV2CSV gets the RegistryV2CSV object from appendTostructuredObjectMap and ensure that there is only one of
+// its kind.
+func (b *bundlev2) getRegistryV2CSV() (*RegistryV2CSV, error) {
+	if csvBasicsIF, ok := b.structuredObjectMap[RegistryV2CSVKind]; ok {
+		if len(*csvBasicsIF) != 1 {
+			return nil, fmt.Errorf("%d %s kind files found in bundle", len(*csvBasicsIF), RegistryV2CSVKind)
+		}
+
+		csvBasics, ok := (*csvBasicsIF)[0].(RegistryV2CSV)
+		if !ok {
+			// Input RegistryV2CSV should have parsed successfully to pass in appendTostructuredObjectMap fxn.
+			// This failure would hav been a programic error.
+			panic("Input is not RegistryV2CSVKind")
+		}
+		return &csvBasics, nil
+	}
+	return nil, fmt.Errorf("the %s resource is not found in bundle", RegistryV2CSVKind)
+}
+
+func (b *bundlev2) getInstallStrategy() (*v1alpha1.NamedInstallStrategy, error) {
+	deploys := b.getDeploytments()
+	var deploySpec appsv1.DeploymentSpec
+	if deploys == nil {
+		// TODO: Do we have to get the deploySpec from depolyment and nothing else?
+		return nil, fmt.Errorf("no deploment is found in bundle")
+	} else if len(*deploys) == 1 {
+		deploySpec = (*deploys)[0].Spec
+	} else {
+		// TODO: Can deployment only be used of CSVs? If not How to find the one that is for install strategy? Can we
```

Comment:
You're saying what if the bundle has more than one deployment object in it? That's a good question - @njhale any thoughts